test/system/src/test/java/io/pravega/test/system/MultiReaderTxnWriterWithFailoverTest.java

```diff
@@ -172,7 +173,7 @@ public void setup() {
         assertTrue(segmentStoreInstance.isRunning());
         log.info("Pravega Segmentstore service instance details: {}", segmentStoreInstance.getServiceDetails());
         //executor service
-        executorService = Executors.newScheduledThreadPool(NUM_READERS + NUM_WRITERS);
+        executorService = Executors.newScheduledThreadPool(NUM_READERS + NUM_WRITERS + 2);
```

Comment:
Not related to this PR, but I don't find this too scalable. Do we really need a dedicated thread for each reader and each writer?

Reply:
Having a separate thread for each of the readers and writers is helpful for debugging tests. This was helpful to find deadlock /thread block issues.